auction-app-backend/src/main/java/com/atlantbh/auctionappbackend/controller/AuthController.java

```diff
@@ -0,0 +1,118 @@
+        this.userService = userService;
+        this.jwtConfig = jwtConfig;
+        this.signAlgorithm = signAlgorithm;
+    }
+
+    @PostMapping(path = "/register")
+    public ResponseEntity<User> processRegister(@RequestBody @Valid User user) {
+        return new ResponseEntity<>(
+                userService.registerUser(user),
+                HttpStatus.OK
+        );
+    }
+
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public Map<String, String> handleMethodArgumentNotValid(MethodArgumentNotValidException ex) {
+        Map<String, String> errors = new HashMap<>();
+        errors.put("status", "error");
+        ex.getBindingResult().getFieldErrors().forEach(error ->
+                errors.put(error.getField(), error.getDefaultMessage()));
+
+        return errors;
+    }
+
+    @GetMapping(path = "/token/refresh")
+    public void refreshToken(
+            HttpServletRequest request,
+            HttpServletResponse response
+    ) throws IOException {
+        String authorizationHeader = request.getHeader(AUTHORIZATION);
+        if (authorizationHeader != null && authorizationHeader.startsWith(jwtConfig.getTokenPrefix())) {
+            try {
+                String refreshToken = authorizationHeader.substring(jwtConfig.getTokenPrefix().length());
+                JWTVerifier verifier = JWT.require(signAlgorithm).build();
+                DecodedJWT decodedJWT = verifier.verify(refreshToken);
+                String username = decodedJWT.getSubject();
+                User user = userService.getUser(username);
+                List<String> authorities = new ArrayList<>();
+                authorities.add(user.getRole());
+                String accessToken = JWT.create()
+                        .withSubject(user.getEmail())
+                        .withExpiresAt(new Date(System.currentTimeMillis() + jwtConfig.getTokenExpAfterMin() * 60 * 1000))
+                        .withIssuer(request.getRequestURL().toString())
+                        .withClaim("roles", authorities)
+                        .sign(signAlgorithm);
+
+                Map<String, String> tokens = new HashMap<>();
+                tokens.put("access_token", accessToken);
+                tokens.put("refresh_token", refreshToken);
+
+                response.setContentType(APPLICATION_JSON_VALUE);
+                new ObjectMapper().writeValue(response.getOutputStream(), tokens);
+
+            } catch (Exception exception) {
+                log.error("Error: " + exception.getMessage());
+                response.setHeader("error", exception.getMessage());
+                response.setStatus(FORBIDDEN.value());
+                Map<String, String> error = new HashMap<>();
+                error.put("error_msg", exception.getMessage());
+                response.setContentType(APPLICATION_JSON_VALUE);
+                new ObjectMapper().writeValue(response.getOutputStream(), error);
```

Comment:
Like for logging. 👍

Also, can we just throw an exception here instead of generating response?

Reply:
In this particular method I've just thrown an exception. But for methods in Filter classes, which implemented dealing with responses like this, throwing an exception doesn't work as expected, so I used the response object, but I cleaned the code a little bit so it isn't clogged up like this.